CMakeLists.txt

```diff
@@ -111,6 +111,23 @@ else()
   cmake_pop_check_state()
 
 endif ()
+
+# Check whether std::float16_t is available and is working
```

Comment:
I see the test failing on fedora:rawhide clang (https://github.com/OSGeo/gdal/actions/runs/13078019777/job/36507595978?pr=11180). I suspect this is due to it having very pedantic warning flags + -Werror that trigger a unused variable for std::float16_t and the arguments of main. So perhaps the following ?

Reply:
Done.